src/app/archive/archive_lib/processor.ml

```diff
@@ -331,6 +331,28 @@ module User_command = struct
            |sql})
         id
 
+    type public_keys = {fee_payer_id: int; source_id: int; receiver_id: int}
+
+    let add_public_keys_if_don't_exist (module Conn : CONNECTION)
```

Comment:
Better name for this or better way to factor out the common code?

Reply:
Could use something like:
```ocaml
let add_pk key = Public_key.add_if_doesn't_exist (module Conn) key in
let%bind fee_payer_id = add_pk (Signed_command.fee_payer_pk t) in
```
but not clear that's better